sdks/python/apache_beam/transforms/core.py

```diff
@@ -1210,6 +1211,30 @@ def __repr__(self):
   def is_default(self):
     return self._is_default
 
+  def to_runner_api(self, context):
+    raise NotImplementedError
+    return beam_runner_api_pb2.WindowingStrategy(
+        window_fn=self.window_fn.to_runner_api(),
+        # TODO(robertwb): Prohibit implicit multi-level merging.
+        merge_status=(MergeStatus.NEEDS_MERGE
+                      if window_fn.is_merging()
+                      else MergeStatus.NON_MERGING),
+        window_coder_id=context.coders.get_id(
+            self.window_fn.get_window_coder()),
+        trigger=self.triggerfn.to_runner_api(context),
+        accumulation_mode=self.accumulation_mode,
+        output_time=self.output_time,
+        closing_behavior=EMIT_ALWAYS,
+        allowed_lateness=0)
+
+  @staticmethod
+  def from_runner_api(self, proto, context):
+    return Windowing(
```

Comment:
Windowing class has different named parameters: windowfn and triggerfn is used there (without underscore).